tests/test_lint.py

```diff
@@ -312,3 +330,62 @@ def test_conda_dockerfile_skip(self):
         expectations = {"failed": 0, "warned": 0, "passed": 0}
         self.assess_lint_status(lint_obj, **expectations)
     
+    def test_pip_no_version_fail(self):
+        """ Tests the pip dependency version definition is present """
+        lint_obj = nf_core.lint.PipelineLint(PATH_WORKING_EXAMPLE)
+        lint_obj.files = ['environment.yml']
+        lint_obj.pipeline_name = 'tools'
+        lint_obj.conda_config = {'name': 'nfcore-tools', 'dependencies': [{'pip': ['multiqc']}]}
+        lint_obj.check_conda_env_yaml()
+        expectations = {"failed": 1, "warned": 0, "passed": 1}
+        self.assess_lint_status(lint_obj, **expectations)
+
+    def test_pip_package_not_latest_warn(self):
+        """ Tests the pip dependency version definition is present """
+        lint_obj = nf_core.lint.PipelineLint(PATH_WORKING_EXAMPLE)
+        lint_obj.files = ['environment.yml']
+        lint_obj.pipeline_name = 'tools'
+        lint_obj.conda_config = {'name': 'nfcore-tools', 'dependencies': [{'pip': ['multiqc=1.4']}]}
+        lint_obj.check_conda_env_yaml()
+        expectations = {"failed": 0, "warned": 1, "passed": 2}
+        self.assess_lint_status(lint_obj, **expectations)
+
+    @mock.patch('requests.get')
+    def test_pypi_timeout_warn(self, mock_get):
+        """ Tests the PyPi connection """
+        # Define the behaviour of the request get mock
+        timeout = requests.exceptions.Timeout()
+        mock_get.side_effect = timeout
+        # Now do the test
+        lint_obj = nf_core.lint.PipelineLint(PATH_WORKING_EXAMPLE)
+        lint_obj.files = ['environment.yml']
+        lint_obj.pipeline_name = 'tools'
+        lint_obj.conda_config = {'name': 'nfcore-tools', 'dependencies': [{'pip': ['multiqc=1.5']}]}
+        lint_obj.check_conda_env_yaml()
+        expectations = {"failed": 0, "warned": 1, "passed": 2}
+        self.assess_lint_status(lint_obj, **expectations)
+
+    @mock.patch('requests.get')
+    def test_pypi_connection_error_warn(self, mock_get):
+        """ Tests the PyPi connection """
+        # Define the behaviour of the request get mock
+        connection_error = requests.exceptions.ConnectionError()
+        mock_get.side_effect = connection_error
```

Comment:
Awesome! Good to learn about this one 👍 

Any reason to assign it to a variable first instead of doing it in one line?

Reply:
Just for educational purpose :D